src/client/components/ToolbarWithGiphy.tsx

```diff
@@ -0,0 +1,199 @@
+      menuItems={[
+        {
+          name: 'giphy-menu',
+          element: (
+            <GiphyMenuWithSearchContext setShowGiphyMenu={setShowGiphyMenu} />
+          ),
+        },
+      ]}
+      button={
+        <StyledGiphtMenuButton
+          onClick={() => setShowGiphyMenu((prev) => !prev)}
+          buttonAction="open-giphy"
+        >
+          <StyledGifIcon />
+        </StyledGiphtMenuButton>
+      }
+      buttonTooltipLabel="Send a Giphy"
+    />
+  );
+}
+
+type GiphyMenuProps = {
+  setShowGiphyMenu: (showGiphyMenu: boolean) => void;
+};
+
+// https://github.com/Giphy/giphy-js/blob/master/packages/react-components/README.md#search-experience
+function GiphyMenuWithSearchContext(props: GiphyMenuProps) {
+  const config = useContext(ConfigContext);
+
+  if (!config?.giphy_api_key) {
+    return null;
+  }
+  return (
+    <SearchContextManager apiKey={config.giphy_api_key}>
+      <GiphyMenu {...props} />
+    </SearchContextManager>
+  );
+}
+
+function GiphyMenu({ setShowGiphyMenu }: GiphyMenuProps) {
+  const { fetchGifs, searchKey } = useContext(SearchContext);
+  // Using the built in composer context to retrieve the attachFilesToComposer function
+  const composerContext = useContext(betaV2.ComposerContext);
+
+  const initialGifsRef = useRef<Promise<GifsResult> | null>(null);
+  const fetchGifsFn = useCallback(
+    (offset: number) => {
+      if (!searchKey) {
+        if (initialGifsRef.current !== null) {
+          return initialGifsRef.current;
+        } else {
+          const result = fetchGifs(offset);
+          initialGifsRef.current = result;
+          return result;
+        }
+      }
+      return fetchGifs(offset);
+    },
+    [fetchGifs, searchKey],
+  );
+
+  const onGiphyClick = useCallback(
+    (gif: IGif, e: React.SyntheticEvent<HTMLElement, Event>) => {
+      e.preventDefault();
+      const gifName = gif.title;
+      const downsizedMed = gif.images.downsized_medium;
+      void fetch(downsizedMed.url)
+        .then((res) => res.blob())
+        .then((blob) => {
+          const file = new File([blob], gifName, {
+            type: 'image/gif',
+          });
+
+          void composerContext?.attachFilesToComposer([file]);
+          setShowGiphyMenu(false);
+        })
+        .catch((error) => {
+          console.error('something went wrong', error);
+          setShowGiphyMenu(false);
+        });
+    },
+    [composerContext, setShowGiphyMenu],
+  );
+
+  return (
+    <>
+      <SearchBar />
+      <StyledGiphyGrid
+        key={searchKey}
+        columns={2}
+        width={300}
+        fetchGifs={fetchGifsFn}
+        onGifClick={onGiphyClick}
+      />
+      <StyledGiphyAttribution
+        src="/static/powered-by-giphy.png"
+        alt="Giphy attribution"
+      />
+    </>
+  );
+}
+
+// The &&& is to increase specificity
+const StyledGiphtMenuButton = styled(betaV2.Button).attrs({
+  className: 'cord-tertiary',
+})`
+  &&& {
+    padding: 4px;
```

Comment:
That much? That must be a bug in our component.

Reply:
It was too confusing to figure out which classes I needed to get this so I just added it 😅